patches/server/0702-Add-BlockBreakBlockEvent.patch

```diff
@@ -19,12 +19,15 @@ index 718c67c630f7799c20df9850d9c2056a7218c220..224ce424b9a22548212ac157ec273c6d
 +            for (net.minecraft.world.item.ItemStack drop : net.minecraft.world.level.block.Block.getDrops(state, world.getMinecraftWorld(), pos, blockEntity)) {
 +                items.add(org.bukkit.craftbukkit.inventory.CraftItemStack.asBukkitCopy(drop));
 +            }
++            Block block = state.getBlock();
 +            io.papermc.paper.event.block.BlockBreakBlockEvent event = new io.papermc.paper.event.block.BlockBreakBlockEvent(org.bukkit.craftbukkit.block.CraftBlock.at(world, pos), org.bukkit.craftbukkit.block.CraftBlock.at(world, source), items);
++            event.setExpToDrop(block.getExpDrop(state, (ServerLevel) world, pos, net.minecraft.world.item.ItemStack.EMPTY, true));
 +            event.callEvent();
 +            for (var drop : event.getDrops()) {
 +                popResource(world.getMinecraftWorld(), pos, org.bukkit.craftbukkit.inventory.CraftItemStack.asNMSCopy(drop));
 +            }
-+            state.spawnAfterBreak(world.getMinecraftWorld(), pos, ItemStack.EMPTY, true);
++            state.spawnAfterBreak(world.getMinecraftWorld(), pos, ItemStack.EMPTY, false);
```

Comment:
I think its best to leave it as true, just because that is what vanilla does. even though that dropExperience parameter does absolutely nothing.

Reply:
See the previous comment, this is supposed to not cause xp to drop.